knimin/lib/data_access.py

```diff
@@ -3191,6 +3186,252 @@ def delete_sample_plate(self, id):
             TRN.add(sql, [id])
             TRN.execute()
 
+    def get_id_by_name(self, field, name):
+        """Converts a field option name to its corresponding ID
+
+        Parameters
+        ----------
+        field : str
+            Field name, i.e., table name under schema "pm"
+        name : str
+            Name of the option
+
+        Returns
+        -------
+        int
+            ID of the option
+
+        Raises
+        ------
+        ValueError
+            If name is not found in the table
+        """
+        with TRN:
+            sql = "SELECT {} FROM {} WHERE name = %s"
+            TRN.add(sql.format(field + '_id', 'pm.' + field), [name])
+            res = TRN.execute_fetchflatten()
+            if not res:
+                raise ValueError('%s is not an option for %s.'
+                                 % (repr(name), repr(field)))
+            return res[0]
+
+    def get_sample_plate_type(self, id=None):
+        """Gets plate type properties of a sample plate
+
+        Parameters
+        ----------
+        id : int, optional
+            ID of an existing sample plate
+            If None, the first plate type will be selected
+
+        Returns
+        -------
+        dict
+            {name : str, cols : int, rows : int, notes : str}
+            Properties of a plate type:
+            Name, number of columns, number of rows, notes
+        """
+        with TRN:
+            sql = """SELECT pm.plate_type.name, cols, rows, pm.plate_type.notes
+                     FROM pm.plate_type
+                     JOIN pm.sample_plate
+                     USING (plate_type_id)
+                     WHERE %s IS NULL OR sample_plate_id = %s
+                     LIMIT 1"""
```

Comment:
Why limit 1? Should only be 1 right? If so, remove the LIMIT. If for whatever reason there is more than 1, we want to be able to detect the problem.